lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -91,7 +91,7 @@ async fn _do_recover_from_snapshot(
 
     log::debug!("Snapshot downloaded to {}", snapshot_path.display());
 
-    let tmp_collection_dir = Path::new(toc.storage_path())
+    let tmp_collection_dir = Path::new(toc.tmp_path())
```

Comment:
If `tmp_path()` returns `Option<&str>`, we can do this:

```suggestion
    let tmp_collection_dir = Path::new(toc.tmp_path().unwrap_or_else(|| toc.storage_path())
```